spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/integration/inbound/PubSubInboundChannelAdapter.java

```diff
@@ -179,6 +181,10 @@ private void logWarning(
     } else {
       LOGGER.warn(re.getMessage());
     }
+    // Log the stacktrace for troubleshoot.
+    StringWriter stringWriter = new StringWriter();
+    re.printStackTrace(new PrintWriter(stringWriter));
```

Comment:
This is not threadsafe -- we probably don't want this here. Logging is the goal.